src/tts_text/borger_dk.py

```diff
@@ -0,0 +1,269 @@
+                urls_to_search=urls_to_search,
+                parsed_urls=parsed_urls,
+                found_urls=found_urls,
+            )
+
+        # We have now followed all the good links in the articles hence
+        # accumulated_articles should now only contain text from the nested articles
+        # on the current page. We add these to the text we have already found on this
+        # page
+        return scraped_text_on_current_page + accumulated_articles, found_urls
+
+    # Recursively follow links and extract articles
+    return follow_links_and_extract_articles(
+        cfg=cfg,
+        category=category,
+        accumulated_articles=accumulated_articles,
+        top_url=url,
+        urls_to_search=urls_to_search,
+        parsed_urls=parsed_urls,
+        found_urls=found_urls,
+    )
+
+
+def follow_links_and_extract_articles(
+    cfg: DictConfig,
+    category: str,
+    accumulated_articles: list[str],
+    top_url: str,
+    urls_to_search: list[str],
+    parsed_urls: list[str],
+    found_urls: list[str],
+) -> tuple[list[str], list[str]]:
+    """Follow links and extract articles.
+
+    Args:
+        cfg: The Hydra configuration object.
+        category: The category of the page.
+        accumulated_articles: The articles that have already been accumulated.
+        top_url: The top URL.
+        urls_to_search: The URLs to search.
+        parsed_urls: The URLs that have already been parsed.
+        found_urls: The URLs that have already been found, but not necessarily
+            parsed.
+
+    Returns:
+        A tuple of the accumulated articles and the found URLs.
+    """
+    desc = f"Extracting articles from {top_url}"
+    parsed_urls.append(top_url)
+    found_urls.extend(urls_to_search + [top_url])
+    for url in tqdm(urls_to_search, desc=desc, leave=False):
+        if url in parsed_urls:
+            continue
+        extracted_articles, found_urls = extract_all_articles(
+            cfg=cfg,
+            url=url,
+            category=category,
+            parsed_urls=parsed_urls,
+            found_urls=found_urls,
+        )
+        accumulated_articles.extend(extracted_articles)
+    return accumulated_articles, found_urls
+
+
+def get_suitable_links(
+    list_link: Tag, category: str, url: str, found_urls: list[str]
+) -> list[str]:
+    """Get suitable links from a list of links.
+
+    Suitable links are links:
+        - to the same category
+        - that are not going up in the hierarchy
+        - that are not refering alternative navigation hierarchies
+        - that are not links to files hosted on borger.dk nor javascript-based
+            navigation links
+        - that have not already been found
+    Args:
+        list_link: The list of links.
+        category: The category of the links.
+        url: The URL of the page where the list of links was found.
+        found_urls: The URLs that have already been found, but not necessarily
+            parsed.
```

Comment:
Same as before with the hanging indent. Maybe it's easier to just use hanging indents throughout docstrings in this file.